R/with.R

```diff
@@ -117,6 +126,25 @@ linters_with_tags <- function(tags, ..., packages = "lintr", exclude_tags = "dep
   modify_defaults(..., defaults = tagged_linters)
 }
 
+#' Create a linter configuration based on all available linters
+#'
+#' @inheritParams linters_with_tags
+#'
+#' @examples
+#' names(all_linters())
+#'
+#' @family linter-config
+#'
+#' @seealso
+#' - [linters_with_defaults] for basing off lintr's set of default linters.
+#' - [linters_with_tags] for basing off tags attached to linters, possibly across multiple packages.
+#' - [available_linters] to get a data frame of available linters.
+#' - [linters] for a complete list of linters available in lintr.
+#' @export
+all_linters <- function(packages = "lintr", ...) {
+  linters_with_tags(tags = NULL, packages = packages, ...)
```

Comment:
What happens in the unlikely case that `tags` is given as a named argument?

Reply:
It will produce an error: 

``` r
library(lintr)
all_linters(tags = "default")
#> Error in linters_with_tags(tags = NULL, packages = packages, ...): formal argument "tags" matched by multiple actual arguments
```

<sup>Created on 2022-12-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

But I doubt anyone will do this because this assumes that the user has looked at the implementation details for this newly introduced function and figured out that `all_linters()` is a wrapper around `linters_with_tags()`, and is deliberately using this argument. But, in that case, they shouldn't be surprised by the error message they get.

Reply:
`modify_defaults(..., defaults = linters_with_tags(tags = NULL, packages = packages)` should not suffer from this bug.